tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,214 @@
+            "Error on line '" + line + "' : only 1 host is allowed in a line."
+        )
+
+    operation = parsed_line.group(1).strip()
+
+    # Get the host and test if it is declared
+    if parsed_line.group(2) is not None:
+        host = parsed_line.group(2).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    elif parsed_line.group(4) is not None:
+        host = parsed_line.group(4).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    else:
+        host = None
+
+    # Get the extra vars and test if they are declared
+    if parsed_line.group(3) is not None:
+        extra_vars = parsed_line.group(3).split(" with ")[1]
+        if extra_vars == "":
+            raise ValueError("Extra vars are required after 'with' keyword.")
+        extra_vars = extra_vars.split(",")
+        extra_vars = [item.strip() for item in extra_vars]
+    else:
+        extra_vars = None
+
+    return (operation, host, extra_vars)
+
+
+@click.command(short_help="Edit Deployment plan.")
+@collections
+@database_dsn
+def edit(
+    collections,
+    database_dsn,
+):
+    with get_session(database_dsn, commit_on_exit=True) as session:
+        planned_deployment_log = get_planned_deployment_log(session)
+        if planned_deployment_log is None:
+            raise click.ClickException(
+                "No planned deployment found, please run `tdp plan` first."
+            )
+
+        operation_list = [
+            (operation.operation, operation.host, operation.extra_vars)
+            for operation in planned_deployment_log.operations
+        ]
+
+        operation_lines = [
+            operation
+            + (" on " + host if host else "")
+            + (" with " + ",".join(extra_vars) if extra_vars else "")
+            for operation, host, extra_vars in operation_list
+        ]
+
+        temp_file = tempfile.NamedTemporaryFile(mode="w", suffix=".csv", delete=False)
+
+        header_text = f"""\
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+# Modify this file to edit the deployment plan.
+# Lines beginning with '#' are comments and will be ignored.
+#
+# Current plan id: {planned_deployment_log.id}
+# Temporary file: {temp_file.name}
+#
+# Execution Order:
+# - Operations are executed in the order they appear below.
+# - You can add, remove, or rearrange operations.
+# - Use `tdp nodes` command to view a list of available operations.
+#
+# Operations options:
+# - Specify a host with the `on` keyword.
+#   Example: hdfs_datanode_config on worker_02
+# - Add extra variables using the `with` keyword and separate with `,`.
+#   Example: wait_sleep with rolling-interval=5, mock-arg=example
+# ------------------- DEPLOYMENT PLAN EDITOR --------------------------
+"""
```

Comment:
We may want to extract it at the top of the file to avoid bloating the code with text.

Reply:
Maybe by defining a function `def get_header_message(plan_id: int, file_name: str)`

Reply:
Done, I created a method, thanks ! ^^